src/main/kotlin/racingcar/domain/GameHandler.kt

```diff
@@ -0,0 +1,70 @@
+package racingcar.domain
+
+import camp.nextstep.edu.missionutils.Randoms
+import racingcar.data.Car
+
+class GameHandler(
+    private val io: IOHandler,
+    private val analyzer: Analyzer,
+) {
+
+    private lateinit var cars: List<Car>
+    private var moveCount: Int = NOT_YET_SET
+
+    fun start() {
+        set()
+        play()
+    }
+
+    private fun set() {
+        setCar()
+        setMoveCount()
+    }
+
+    private fun setCar()  {
+        io.show(SENTENCE_FOR_GETTING_NAME, true)
+
+        cars = io.getCarNames().map { name ->
+            Car(name)
+        }
+    }
+
+    private fun setMoveCount() {
+        io.show(SENTENCE_FOR_GETTING_MOVEMENT, true)
+
+        moveCount = io.getMoveCount()
+    }
+
+    private fun play() {
+        io.show(EMPTY_SENTENCE_FOR_LINE_BREAK, true)
+        io.show(SENTENCE_FOR_RESULT, true)
+
+        repeat(moveCount) {
+            moveCars(cars)
+
+            io.showLocation(cars)
+            io.show(EMPTY_SENTENCE_FOR_LINE_BREAK, true)
+        }
+        io.showWinner(analyzer.judgeWinner(cars))
+    }
+
+    private fun moveCars(cars: List<Car>) {
+        for (car in cars) {
+            if (analyzer.isMoveAllowed(getRandomNum())) {
+                car.move()
+            }
+        }
```

Comment:
[1] 성능과 메모리 면에서 많이 부족하지만, 
if문을 사전에 처리하는 느낌으로 indent를 줄일 수 있긴 했네요,,, ㅎㅎ
```suggestion
        val randNumberList = listOf(analyzer.isMoveAllowed(getRandomNum()))
        val indicesToMove = randNumberList.mapIndexedNotNull { index, value ->
            index.takeIf { value } ?: null
        }

        for (index in indicesToMove) {
            cars[index].move()
        }
```

Reply:
조건에 해당하는 index를 사전에 필터링하는 아이디어가 참신한 것 같아요. 좀 더 적합한 사용 사례가 있을 것 같은데, 떠오르지 않네요😅